### PR TITLE
feat: limit stars list to only the last 3 unused stars

### PR DESCRIPTION
**Pull request by bdougie**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds the ability to limit stars to only the last 3 unused stars. 

## Related Tickets & Documents

https://github.com/open-sauced/open-sauced/issues/1096
## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [x] 📓 docs
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

- Fix the supabase DB
- Centralize user star contributions

### CODE REVIEW

**Comment by bdougie** [comment]
Tests are passing, but there are a few things to consider prior to merging. This will break how all user goal repos interact with the open sauced project.
- [ ] Docs explaining how the stars are rendering now and data is collected.
- [ ] dependent on the above, I will need to create a second admin action (update-workflows.js) update all goals-caching.yml to include the open-sauced actions.
- [ ] Remove the persisted query for FetchiViewerStars

File: stories/1-Button.stories.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const primary = () => (
 
 export const PrimaryWithText = () => (
   <div>
-    <h2 style={{fontFamily: "Roboto", color:"grey", fontSize: "1.0rem"}}>Set goals directly on Github</h2>
+    <h2 style={{fontFamily: "Roboto", color: "grey", fontSize: "1.0rem"}}>Set goals directly on Github</h2>
```
Comment (mtfoley) [inline stories/1-Button.stories.js:19]: Should this use @storybook/addon-actions like the other story buttons do?

File: src/components/RepositoryGoals.js [hunk 2]
```diff
@@ -67,7 +49,10 @@ function RepositoryGoals({user}) {
   };
 
   const data = repository && repository.data && repository.data.text && JSON.parse(repository.data.text);
+  const viewerStars = repository && repository.stars && repository.stars.text && JSON.parse(repository.stars.text);
 
+  const stars = remainingStars(data, viewerStars);
+  console.log(stars);
```
Comment (bdougie) [inline src/components/RepositoryGoals.js:55]:
```suggestion
```

File: stories/1-Button.stories.js [hunk 3]
```diff
@@ -34,8 +34,7 @@ export const secondaryWithImageAndText = () => (
 
 export const addTo = () => (
   <div>
-    {/* eslint-disable-next-line no-undef */}
-    <a onClick={() => _handleGoalCreation(goal.nameWithOwner)} href="#">
+    <a onClick={() => console("goal creation simulated")} href="#">
```
Comment (mtfoley) [inline stories/1-Button.stories.js:37]: Meant this line for using @storybook/addon-actions like the other story buttons do

**Comment by bdougie** [comment]
This is ready to merge once I catch a few more errors

https://github.com/open-sauced/actions/runs/3387161679?check_suite_focus=true

**Comment by 0-vortex** [comment]
> This is ready to merge once I catch a few more errors in the action
> 
> https://github.com/open-sauced/actions/runs/3387161679?check_suite_focus=true

"being your own QA" rules

**Pull request merged**